src/hooks/useUndernames/useUndernames.tsx

```diff
@@ -0,0 +1,263 @@
+import { ColumnType } from 'rc-table/lib/interface';
+import { useEffect, useState } from 'react';
+
+import { useArweaveCompositeProvider, useIsMobile } from '..';
+import {
+  CalendarTimeIcon,
+  ChevronUpIcon,
+  NotebookIcon,
+  PencilIcon,
+  TargetIcon,
+  TrashIcon,
+} from '../../components/icons/index';
+import CopyTextButton from '../../components/inputs/buttons/CopyTextButton/CopyTextButton';
+import {
+  ArweaveTransactionID,
+  PDNTContractJSON,
+  UndernameMetadata,
+} from '../../types';
+import eventEmitter from '../../utils/events';
+
+export function useUndernames(id?: ArweaveTransactionID) {
```

Comment:
why is this optional?

Reply:
was rapid fixing and didnt think about it, removing now

Reply:
removed